jgrapht-core/src/main/java/org/jgrapht/alg/drawing/TwoLayeredBipartiteLayout2D.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * (C) Copyright 2020-2020, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.drawing;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+import org.jgrapht.Graph;
+import org.jgrapht.alg.drawing.model.Box2D;
+import org.jgrapht.alg.drawing.model.LayoutModel2D;
+import org.jgrapht.alg.drawing.model.Point2D;
+import org.jgrapht.alg.interfaces.PartitioningAlgorithm.Partitioning;
+import org.jgrapht.alg.partition.BipartitePartitioning;
+import org.jgrapht.alg.util.Pair;
+
+/**
+ * A two layered bipartite layout.
+ * 
+ * The algorithm draws a bipartite graph using straight edges. Vertices are arranged along two
+ * vertical or horizontal lines. No attempt is made to minimize edge crossings.
+ * 
+ * The order of the vertices can be adjusted by providing a vertex comparator. Similarly the user
+ * can also determine the two partitions or can let the algorithm compute them.
+ * 
+ * @author Dimitrios Michail
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class TwoLayeredBipartiteLayout2D<V, E>
+    implements
+    LayoutAlgorithm2D<V, E>
+{
+    protected Comparator<V> vertexComparator;
+    protected boolean vertical;
+    protected Set<V> partition;
+
+    /**
+     * Create a new layout
+     */
+    public TwoLayeredBipartiteLayout2D()
```

Comment:
Can we have a normal constructor too instead of just the builder pattern?
`TwoLayeredBipartiteLayout2D(Set<V> partition, Comparator<V> vertexComparator, boolean vertical )`